### PR TITLE
Support Lustre 2.12

### PR DESCRIPTION
**Pull request by jgrund**
Add support for Lustre 2.12

Signed-off-by: Joe Grund <jgrund@whamcloud.io>

### CODE REVIEW

**Comment by jgrund** [comment]
Besides test fixes, we should also consider if this patch still works with Lustre 2.10.x

**Comment by jgrund** [comment]
Haven't dug into the cause but seeing this after adding a server:

screen shot 2019-01-31 at 4 04 54 pm


Usually a network will be selected, in this case it's all unselected.

File: chroma_agent/device_plugins/linux_network.py [hunk 1]
```diff
@@ -233,7 +233,7 @@ class LNetNid:
     def __init__(self, lnet_nis_line, interfaces):
         tokens = lnet_nis_line.split()
 
-        assert len(tokens) == 9
+        assert len(tokens) == 9, "line: "+lnet_nis_line
```
Comment (jgrund) [inline chroma_agent/device_plugins/linux_network.py]: Leftover?

File: chroma_agent/action_plugins/manage_targets.py [hunk 2]
```diff
@@ -1066,7 +1066,17 @@ def convert_targets(force=False):
...
+        [
+            "crm_attribute",
+            "--type",
+            "crm_config",
+            "--name",
+            "maintenance-mode",
+            "--update",
+            "true",
+        ]
+    )
```
Comment (jgrund) [inline chroma_agent/action_plugins/manage_targets.py]: Why switch away from pcs in this patch?

File: chroma_agent/action_plugins/manage_targets.py [hunk 3]
```diff
@@ -1098,7 +1108,16 @@ def convert_targets(force=False):
     for wait in wait_list:
         console_log.info("Waiting on %s", wait[0])
         _wait_target(*wait)
-    AgentShell.run(["pcs", "property", "unset", "maintenance-mode"])
+    AgentShell.try_run(
```
Comment (jgrund) [inline chroma_agent/action_plugins/manage_targets.py]: Why switch away from pcs in this patch?

**Comment by jgrund** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/whamcloud/iml-agent) found an issue: No exception type(s) specified

It's currently on:
[chroma_agent\device_plugins\audit\lustre\__init__.py:140
](https://github.com/whamcloud/iml-agent/blob/f82104164e268b01f492f99c6ca3992ea62da2a6/chroma_agent/device_plugins/audit/lustre/__init__.py#L140)Commit f82104164e268b01f492f99c6ca3992ea62da2a6

**Comment by jgrund** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/whamcloud/iml-agent) found an issue: No exception type(s) specified

It's currently on:
[chroma_agent\device_plugins\audit\lustre\__init__.py:631
](https://github.com/whamcloud/iml-agent/blob/ade00a42deac1f78ed8f2ebb9d0a7c1f817b5b7f/chroma_agent/device_plugins/audit/lustre/__init__.py#L631)Commit ade00a42deac1f78ed8f2ebb9d0a7c1f817b5b7f

**Comment by jgrund** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/whamcloud/iml-agent) found an issue: Try, Except, Pass detected.

It's currently on:
[chroma_agent\device_plugins\audit\lustre\__init__.py:260
](https://github.com/whamcloud/iml-agent/blob/f82104164e268b01f492f99c6ca3992ea62da2a6/chroma_agent/device_plugins/audit/lustre/__init__.py#L260)Commit f82104164e268b01f492f99c6ca3992ea62da2a6

**Comment by jgrund** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/whamcloud/iml-agent) found an issue: No exception type(s) specified

It's currently on:
[chroma_agent\device_plugins\audit\lustre\__init__.py:194
](https://github.com/whamcloud/iml-agent/blob/f82104164e268b01f492f99c6ca3992ea62da2a6/chroma_agent/device_plugins/audit/lustre/__init__.py#L194)Commit f82104164e268b01f492f99c6ca3992ea62da2a6

**Comment by jgrund** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/whamcloud/iml-agent) found an issue: No exception type(s) specified

It's currently on:
[chroma_agent\device_plugins\audit\lustre\__init__.py:537
](https://github.com/whamcloud/iml-agent/blob/f82104164e268b01f492f99c6ca3992ea62da2a6/chroma_agent/device_plugins/audit/lustre/__init__.py#L537)Commit f82104164e268b01f492f99c6ca3992ea62da2a6

**Comment by jgrund** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/whamcloud/iml-agent) found an issue: No exception type(s) specified

It's currently on:
[chroma_agent\device_plugins\audit\lustre\__init__.py:260
](https://github.com/whamcloud/iml-agent/blob/f82104164e268b01f492f99c6ca3992ea62da2a6/chroma_agent/device_plugins/audit/lustre/__init__.py#L260)Commit f82104164e268b01f492f99c6ca3992ea62da2a6

**Comment by jgrund** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/whamcloud/iml-agent) found an issue: No exception type(s) specified

It's currently on:
[chroma_agent\device_plugins\audit\lustre\__init__.py:480
](https://github.com/whamcloud/iml-agent/blob/f82104164e268b01f492f99c6ca3992ea62da2a6/chroma_agent/device_plugins/audit/lustre/__init__.py#L480)Commit f82104164e268b01f492f99c6ca3992ea62da2a6

**Comment by jgrund** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/whamcloud/iml-agent) found an issue: No exception type(s) specified

It's currently on:
[chroma_agent\device_plugins\audit\lustre\__init__.py:154
](https://github.com/whamcloud/iml-agent/blob/f82104164e268b01f492f99c6ca3992ea62da2a6/chroma_agent/device_plugins/audit/lustre/__init__.py#L154)Commit f82104164e268b01f492f99c6ca3992ea62da2a6

**Comment by jgrund** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/whamcloud/iml-agent) found an issue: No exception type(s) specified

It's currently on:
[chroma_agent\device_plugins\audit\lustre\__init__.py:336
](https://github.com/whamcloud/iml-agent/blob/f82104164e268b01f492f99c6ca3992ea62da2a6/chroma_agent/device_plugins/audit/lustre/__init__.py#L336)Commit f82104164e268b01f492f99c6ca3992ea62da2a6

File: chroma_agent/device_plugins/linux_network.py (hunk 1 above)
Comment (utopiabound) [reply chroma_agent/device_plugins/linux_network.py]: yes... but I'm not sure it's not useful.

**Comment by utopiabound** [comment]
> Haven't dug into the cause but seeing this after adding a server:
> 
> screen shot 2019-01-31 at 4 04 54 pm
> 
> Usually a network will be selected, in this case it's all unselected.

Is this Montitored vs. Managed?

**Comment by jgrund** [comment]
> Is this Monitored or Managed?

Managed

File: chroma_agent/action_plugins/manage_targets.py (hunk 3 above)
Comment (utopiabound) [reply chroma_agent/action_plugins/manage_targets.py]: This got rolled in, as extra, this really isn't related to this issue.

Comment (jgrund) [reply chroma_agent/action_plugins/manage_targets.py]: Possible this is causing issues in SSI?

**Comment by jgrund** [comment]
Landing blocked on resolution to: https://jira.whamcloud.com/browse/LU-11947

File: tests/data/lustre_versions/dump_lustre_proc [hunk 4]
```diff
@@ -109,10 +132,12 @@ puts "Merging with /proc testdata in: #{lustre_version}/proc"
...
+LUSTRE_LNET_SEARCH.each do |root|
   dump_directory(root) do |entry|
-    dump_entry(entry, dump_path)
-    dump_entry(entry, test_path, false) if test_entry?(entry)
+    dump_param(entry, dump_path, root)
+    dump_param(entry, test_path, root, false) if test_entry?(entry)
   end
 end
 
+# for d in sys/kernel/debug/lustre "sys/fs/lustre" proc/sys/lnet proc/fs/lustre ; do for i in $(cd $d; find * -type f); do f=$(echo $i|tr / .); git mv $d/$i $f; done ;done
```
Comment (johnsonw) [inline tests/data/lustre_versions/dump_lustre_proc]: Can this be removed?

**Pull request merged**